core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -151,19 +151,22 @@ public static HDPath m(ChildNumber... children) {
      * Where a letter "H" means hardened key. Spaces are ignored.
      */
     public static HDPath parsePath(@Nonnull String path) {
-        String[] parsedNodes = path.replace("M", "").split("/");
+        boolean hasPrivateKey = path.startsWith(PREFIX_PRIVATE);
+        String[] parsedNodes = path.replace(PREFIX_PUBLIC, "")
```

Comment:
Actually I think the whole .replace() is wrong. Only a leading `M` or `m` should be removed. Later occurences would be illegal and should lead to some form of error. (I'm aware that the previous implementation had this problem too.)